select.go

```diff
@@ -152,22 +167,33 @@ func (s *Select) OnChange(key rune, config *PromptConfig) bool {
 	// and we have modified the filter then we should move the page back!
 	opts, idx := paginate(pageSize, options, s.selectedIndex)
 
+	tmplData := SelectTemplateData{
+		Select:        *s,
+		SelectedIndex: idx,
+		ShowHelp:      s.showingHelp,
+		PageEntries:   opts,
+		Config:        config,
+	}
+
 	// render the options
-	s.Render(
-		SelectQuestionTemplate,
-		SelectTemplateData{
-			Select:        *s,
-			SelectedIndex: idx,
-			ShowHelp:      s.showingHelp,
-			PageEntries:   opts,
-			Config:        config,
-		},
-	)
+	s.Render(SelectQuestionTemplate, tmplData)
+
+	offset := computeCursorOffset(SelectQuestionTemplate, tmplData, opts, idx, s.termWidthSafe())
+	s.OffsetCursor(offset)
 
 	// keep prompting
 	return false
 }
 
+func (s *Select) width() int {
```

Comment:
This looks almost exactly like `Renderer.width` - any way we could add an argument to the function that passes the number of characters?

Reply:
oops, that's vestigial; I'll just delete it.